app/controllers/katello/concerns/organizations_controller_extensions.rb

```diff
@@ -29,6 +29,7 @@ def create_with_dynflow
             @taxonomy = Organization.new(resource_params)
             sync_task(::Actions::Katello::Organization::Create, @taxonomy)
             @taxonomy.reload
+            session[:organization_id] = @taxonomy.id
```

Comment:
I mean `taxonomy_class.current = @taxonomy` that is missing.

> it redirects you back to /organizations/new

Agreed. We could change the behavior of `#select` a bit to redirect to a location stored in the session (by using `#saved_redirect_url_or`), but it's up to you. I just want to prevent code duplication between Katello controllers and those in Foreman.
Other option would be refactoring the controller in Foreman and extracting the functionality to a service class, that could be called from here.